plugins/tech-insights-backend-module-jsonfc/src/service/JsonRulesEngineFactChecker.test.ts

```diff
@@ -156,17 +185,22 @@ describe('JsonRulesEngineFactChecker', () => {
         'Failed to run rules engine, Undefined fact: somefact',
       );
     });
-    it('should respond with result, facts, fact schemas and checks', async () => {
-      latestFactsByIdsMock.mockImplementation(() =>
-        Promise.resolve({
-          ['test-factretriever']: {
-            id: 'test-factretriever',
-            facts: {
-              testnumberfact: 3,
-            },
-          },
+
+    it('should skip checks where fact data is missing', async () => {
+      const skipped = async () =>
+        await factChecker.runChecks('a/a/a', ['brokennotfound']);
```

Comment:
Do you need to have this additional async/await here and in other places?

```suggestion
      const skipped = factChecker.runChecks('a/a/a', ['brokennotfound']);
```